packages/other/src/FlexInput.ts

```diff
@@ -0,0 +1,127 @@
+import { HTMLWidget } from "@hpcc-js/common";
+
+export class FlexInput extends HTMLWidget {
+    _span;
+    _input;
+    _wrapper;
+    constructor() {
+        super();
+    }
+    enter(domNode, element) {
+        super.enter(domNode, element);
+        this._wrapper = element.append("div")
+            .attr("class", "flex-input-wrapper")
+            .style("position", "relative")
+            ;
+        this._input = element.append("input")
+            .attr("class", "flex-input-input")
+            .style("position", "absolute")
+            .style("display", "inline-block")
+            ;
+        this._span = element.append("span")
+            .attr("class", "flex-input-span")
+            .style("position", "absolute")
+            .style("display", "inline-block")
+            ;
+    }
+    update(domNode, element) {
+        super.update(domNode, element);
+
+        const isCollapsed = this.lockCollapse() || this.collapsed() || this.value() !== "";
+
+        if (isCollapsed) {
+            this.collapseLabel();
+        } else {
+            this.uncollapseLabel();
+        }
+
+        this._span
+            .style("font-family", this.fontSize() + "px")
+            .style("transition", "all 400ms")
+            .text(this.label())
+            ;
+        this._input
+            .style("font-family", this.fontSize() + "px")
+            .style("box-sizing", "border-box")
+            .style("height", this.padding() + this.fontSize() + this.collapsedFontSize() + "px")
+            ;
+        this._input.node().value = this.value();
```

Comment:
Should be `this._input.prop("value", this.value())` (I think `value` is a prop and not an attr?)